Packs/IntSight/Integrations/Rapid7ThreatCommand/Rapid7ThreatCommand.py

```diff
@@ -0,0 +1,5354 @@
+    """
+
+    return reputation_handler(
+        args, client, url_reputation_handler, IOCType.URL.value.lower()
+    )
+
+
+@polling_function(
+    name="domain",
+    interval=arg_to_number(demisto.args().get("interval_in_seconds", DEFAULT_INTERVAL)),
+    timeout=arg_to_number(demisto.args().get("timeout_in_seconds", DEFAULT_TIMEOUT)),
+    requires_polling_arg=False,
+)
+def domain_command(args: dict[str, Any], client: Client) -> PollResult:
+    """
+    Enrich domain IOC (Generic reputation command).
+
+    Args:
+        client (Client): Threat Command API client.
+        args (Dict[str, Any]): Command arguments from XSOAR.
+
+    Returns:
+        PollResult: outputs, readable outputs and raw response for XSOAR.
+    """
+    return reputation_handler(
+        args, client, domain_reputation_handler, IOCType.DOMAIN.value.lower()
+    )
+
+
+def reputation_handler(
+    args: dict[str, Any], client: Client, handler_command: Callable, key: str
+) -> PollResult:
+    """
+    Handle with all reputation commands.
+
+    Args:
+        client (Client): Threat Command API client.
+        args (Dict[str, Any]): Command arguments from XSOAR.
+        handler_command (Callable): Handler command for each command.
+        key (str): Key for the IOC.
+
+    Returns:
+        PollResult: outputs, readable outputs and raw response for XSOAR.
+    """
+    execution_metrics = ExecutionMetrics()
+    execution_metrics.general_error += 1
+    ioc_values: List[str] = argToList(args[key])
+    current_file = ioc_values[0]
+    response = client.enrich_ioc(ioc_value=current_file)
+    command_results = []
+    if response["Status"] == "QuotaExceeded":
+        execution_metrics.quota_error += 1
+        command_results.append(
+            CommandResults(
+                readable_output=ReadableErrors.ENRICH_FAIL.value.format(
+                    response["Status"]
+                )
+            )
+        )
+        command_results.append(cast(CommandResults, execution_metrics.metrics))
+        return PollResult(
+            response=command_results,
+            continue_to_poll=False,
+        )
+
+    if response["Status"] == "Failed":
+        execution_metrics.general_error += 1
+        ioc_values.remove(current_file)
+        command_results.append(
+            CommandResults(
+                readable_output=ReadableErrors.ENRICH_FAIL.value.format(
+                    response["Status"]
+                )
+            )
+        )
+        command_results.append(cast(CommandResults, execution_metrics.metrics))
+        if not ioc_values:
+            return PollResult(
+                response=command_results,
+                continue_to_poll=False,
+            )
+
+        args[key] = (",").join(ioc_values)
+        return PollResult(
+            response=command_results,
+            args_for_next_run=args,
+            continue_to_poll=True,
+        )
+    if response["Status"] == "Done":
+        execution_metrics.success += 1
```

Comment:
we want to increase by 1 also when the response is successful but status is not done yet

Reply:
Right now I increase by 1 when the process is finished. If I increase it by 1 when the status is InProgress, it might run a few times and increase it by 1.

Reply:
in progress is a successful state - the goal is to count how many API calls were made in total, not just the last one.

Reply:
Should I increase the `execution_metrics.success`?

Reply:
yes